test/test_client_async.py

```diff
@@ -0,0 +1,583 @@
+import ctypes
+import struct
+import unittest
+import logging
+import time
+import mock
+
+from datetime import datetime
+from subprocess import Popen
+from os import path, kill
+import snap7
+from snap7.snap7exceptions import Snap7Exception
+from snap7.snap7types import S7AreaDB, S7WLByte, S7DataItem
+from snap7 import util
+
+
+logging.basicConfig(level=logging.WARNING)
+
+ip = '127.0.0.1'
+tcpport = 1102
+db_number = 1
+rack = 1
+slot = 1
+
+
+class TestClient(unittest.TestCase):
```

Comment:
it looks like you copied the whole test suite for the client module?

This is not required. You can only copy and modify the tests that are actually overriden in the class.

Reply:
Fixed